cmd/ipfs/init.go

```diff
@@ -49,19 +48,17 @@ var initCmd = &cmds.Command{
 			res.SetError(err, cmds.ErrNormal)
 			return
 		}
+
 		if !bitsOptFound {
 			nBitsForKeypair = nBitsForKeypairDefault
 		}
 
-		rpipe, wpipe := io.Pipe()
-		go func() {
-			defer wpipe.Close()
-			if err := doInit(wpipe, req.Context().ConfigRoot, force, nBitsForKeypair); err != nil {
-				res.SetError(err, cmds.ErrNormal)
-				return
-			}
-		}()
-		res.SetOutput(rpipe)
+		var buf bytes.Buffer
+		if err := doInit(&buf, req.Context().ConfigRoot, force, nBitsForKeypair); err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		res.SetOutput(&buf)
```

Comment:
> The implementation of io.Pipe still uses a byte slice to buffer when it can't write through immediately. If you really wan't this you would want an os.Pipe instead, which uses the pipe syscall to get two FDs from the kernel.

Do you mean that io errors will not propagate using an `io.Pipe`? i haven't encountered this, usually all errors are fwded correctly, but i havent tried to test extensively.

Btw, this used to be a buffer before, and i kept encountering problems, so i switched it to a pipe. Those problems seemed to have gone away (tests pass) or weren't tested for. I'm wary that switching it to a buffer will cause untested failures.

> IMHO the pipe makes the code convoluted without any gain over that.

Hm, in this case I think the opposite-- the buffer makes it more convoluted because now i have to worry about data sitting in this buffer i've allocated and growing, instead of what is supposed to work as a simple io connector.

I mention this because a codebase as large as ours will rarely find aesthetic agreement. We are all likely to have different and conflicting subjective aesthetic preferences. In cases like these we should make decisions based on surface area of problems, readability of the code, correctness, etc. Aesthetics is just going to drive us all crazy, so i don't think it's a good thing to discuss (think tabs vs spaces).

In this particular case, my view is: 

- surface area, pipe is a bit more surface area, because there's two variables. it's not very significant though.
- readability isn't very different, and we have different people on different sides (i think pipe is much cleaner, you think buffer is much cleaner)
- correctness, i believe `io.Pipe` is more correct (error propagation), but i'm not 100% sure.

I dont really care, if you're absolutely sure it's equivalent, go for it. :smile: 

---

@whyrusleeping lol cant tell if trolling or serious.